homeassistant/components/tellduslive/__init__.py

```diff
@@ -210,23 +203,19 @@ def __init__(self, hass, config, session):
 
         self._hass = hass
         self._config = config
-
-        self._interval = config.get(DOMAIN, {}).get(
-            CONF_UPDATE_INTERVAL, DEFAULT_UPDATE_INTERVAL)
-        _LOGGER.debug('Update interval %s', self._interval)
         self._client = session
 
-    def update(self, *args):
-        """Periodically poll the servers for current state."""
-        _LOGGER.debug('Updating')
-        try:
-            self._sync()
-        finally:
-            track_point_in_utc_time(
-                self._hass, self.update, utcnow() + self._interval)
+        interval = config.get(DOMAIN, {}).get(CONF_UPDATE_INTERVAL,
+                                              DEFAULT_UPDATE_INTERVAL)
+        _LOGGER.debug('Update interval %s', interval)
+        async_track_time_interval(self._hass, self.update, interval)
```

Comment:
Revert this. We're in sync context here. Also, we should probably move it outside the `__init__`, since we don't want side effects in the constructors. So create the client, then call `track_time_interval(hass, client.update, interval)`